modules/siddhi-core/src/main/java/io/siddhi/core/event/ComplexEventChunk.java

```diff
@@ -35,6 +37,16 @@
     protected E last;
     protected boolean isBatch = false;
 
+    public List<E> toList() {
```

Comment:
Why are we converting ComplexEventChunk to list, its an unnecessary cost during execution. 
Try to use the ComplexEventChunk itself